app/models/basics/Implicits.scala

```diff
@@ -15,8 +15,19 @@ import scala.language.implicitConversions
 object Implicits extends Implicits
 
 trait Implicits {
-  implicit def authedRequestToDBAccess(implicit request: UserAwareRequest[_]): DBAccessContext = {
-    DBAccessContext(request.userOpt)
+  implicit def userAwareRequestToDBAccess(implicit request: UserAwareRequest[_]): DBAccessContext = {
+    DBAccessContext(request.identity)
+  }
+
+  implicit def securedRequestToDBAccess(implicit request: SecuredRequest[_]): DBAccessContext = {
+    DBAccessContext(Some(request.identity))
+  }
+  //implicit causes ambiguous implicit values
```

Comment:
why does it become ambiguous? is there nothing we can do?

Reply:
I’ve looked into this today, and I feel like I’ve gotten close with https://docs.scala-lang.org/tutorials/FAQ/chaining-implicits.html but it still doesn’t quite work… I see the issue now and if it’s not easy then explicit conversion is ok too

Reply:
does it belong in another file because it is not implicit?

Reply:
No I think its fine here, it’s still about the implicit request… also, I’ll try to do this implicitly one more time tomorrow with fresh eyes.